doc/syslog/syslog-rate-limit-design.md

```diff
@@ -0,0 +1,376 @@
+
+Example:
+config syslog rate-limit-host --interval 300 --burst 20000
+config syslog rate-limit-host --interval 300
+config syslog rate-limit-host --burst 20000
+config syslog rate-limit-container bgp --interval 300 --burst 20000
+```
+
+> Note: set interval or burst to 0 will disable rate limit.
+
+Show rate limit:
+
+```
+show syslog rate-limit-host
+show syslog rate-limit-container [<service_name>]
+
+Example:
+show syslog rate-limit-host
+INTERVAL     BURST
+----------   --------
+500          50000
+
+show syslog rate-limit-container
+SERVICE    INTERVAL     BURST
+--------   ----------   --------
+bgp        500          N/A
+snmp       300          20000
+swss       2000         12000
+
+show syslog rate-limit-container bgp
+SERVICE    INTERVAL     BURST
+--------   ----------   --------
+bgp        500          5000
+```
+
+#### YANG model
+
+```yang
+...
+container SYSLOG_CONFIG {
+
+    description "SYSLOG_CONFIG part of config_db.json";
+
+    container GLOBAL {
+        leaf rate_limit_interval {
+            description "Message rate limit interval";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+
+        leaf rate_limit_burst {
+            description "Message rate limit burst";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+    }
+    /* end of list SYSLOG_CONFIG_LIST */
+}
+/* end of container SYSLOG_CONFIG */
+
+container SYSLOG_CONFIG_FEATURE {
+
+    description "SYSLOG_CONFIG_FEATURE part of config_db.json";
+
+    list SYSLOG_CONFIG_FEATURE_LIST {
+
+        key "service";
+
+        leaf service {
+            description "Service name";
+            type leafref {
+                path "/feature:sonic-feature/feature:FEATURE/feature:FEATURE_LIST/feature:name";
+            }
+        }
+
+        leaf rate_limit_interval {
+            description "Message rate limit interval";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+
+        leaf rate_limit_burst {
+            description "Message rate limit burst";
+            type uint32 {
+                range 0..2147483647;
+            }
+        }
+    }
+    /* end of list SYSLOG_CONFIG_FEATURE_LIST */
+}
+/* end of container SYSLOG_CONFIG_FEATURE */
+...
+
+
+container FEATURE {
+
+    description "feature table in config_db.json";
+
+    list FEATURE_LIST {
+        ...
+
+        leaf support_syslog_rate_limit {
+            description "This configuration indicates if the feature support configuring syslog rate limit";
+            type stypes:boolean_type;
+            default "false";
+        }
+
+        leaf has_customized_syslog_rate_limit {
+            description "This configuration indicates if the feature has customized syslog rate limit configuration";
+            type stypes:boolean_type;
+            default "false";
+        }
+
+        ...
+    }
+}
+```
+
+### Warmboot and Fastboot Design Impact
+
+N/A
```

Comment:
There might be some impact for fastboot, please add TBD

Reply:
Sure